Source/JavaScriptCore/heap/BlockDirectory.cpp

```diff
@@ -67,19 +67,17 @@ void BlockDirectory::updatePercentageOfPagedOutPages(SimpleStats& stats)
     // For some reason this can be unsigned char or char on different OSes...
     using MincoreBufferType = std::remove_pointer_t<FunctionTraits<decltype(mincore)>::ArgumentType<2>>;
     static_assert(std::is_same_v<std::make_unsigned_t<MincoreBufferType>, unsigned char>);
-    // pageSize is effectively a constant so this isn't really variable.
-    IGNORE_CLANG_WARNINGS_BEGIN("vla")
-    MincoreBufferType pagedBits[numberOfPagesInMarkedBlock];
-    IGNORE_CLANG_WARNINGS_END
+    Vector<MincoreBufferType, 16> pagedBits(numberOfPagesInMarkedBlock, MincoreBufferType { });
 
     for (auto* handle : m_blocks) {
         if (!handle)
             continue;
 
-        auto markedBlockSizeInBytes = static_cast<size_t>(reinterpret_cast<char*>(handle->end()) - reinterpret_cast<char*>(handle->start()));
+        char* pageStart = reinterpret_cast<char*>(handle->atomAt(0));
```

Comment:
Instead of using `atomAt(0)`, can you add a `MarkedBlock::Handle::pageStart()` method that returns the value of `reinterpret_cast<char*>(atomAt(0))`?  I think that would be clearer and would have avoided this fall out when we changed the position of the header/footer in the MarkedBlock.

Have you also grepped the code for uses of handle->start() to make sure that there're no other instances of this bug?

Reply:
Sounds good.